File: IT/Containers/Esercizi/Java/Esercizio23.java
/* file Esercizio23.java */

import java.util.SortedSet;
import java.util.TreeSet;

public class Esercizio23 {
	public static void main(String [] args) {
		SortedSet<Integer> a = new TreeSet<Integer>();

		a.add(0);
		a.add(1);
		a.add(2);
		a.add(3);
		a.add(4);

		SortedSet<Integer> b = new TreeSet<Integer>();

		b.add(1);
		b.add(2);
		b.add(3);

		System.out.println("A = " + a);
		System.out.println("B = " + b);

		/* se tutti gli elementi di B sono anche elementi di A, allora B è un sottoinsieme di A */

		if (a.containsAll(b)) {
			System.out.println("L'insieme B è un sottoinsieme di A!");
		}
	}
}

File: IT/1/Containers/Esercizi/C++/esercizio23.cpp
/*
file esercizio23.cpp
compilare con: g++ -std=c++11 esercizio23.cpp
*/

#include <iostream>
#include <iterator>
#include <algorithm>
#include <set>

using std::cout;
using std::endl;
using std::ostream_iterator;
using std::set;
using std::includes;

int main()
{
	set<int> a = {0, 1, 2, 3, 4};
	set<int> b = {1, 2, 3};

	cout << "A = [";
	copy(a.begin(), a.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	cout << "B = [";
	copy(b.begin(), b.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	/* se tutti gli elementi di B sono anche elementi di A, allora B è un sottoinsieme di A */

	if (includes(a.begin(), a.end(), b.begin(), b.end())) {
		cout << "L'insieme B è un sottoinsieme di A!" << endl;
	}
}

